Guard dialog.showModal() and "check popover validity" with fully active checks

maraisr opened the pull request:
#### 13c237d80ef42a99d843f7de2225994ddef18a16
<pre>
Guard dialog.showModal() and &quot;check popover validity&quot; with fully active checks
<a href="https://bugs.webkit.org/show_bug.cgi?id=281550">https://bugs.webkit.org/show_bug.cgi?id=281550</a>

Reviewed by Darin Adler.

There was a WHATWG/html spec change to now guard showModal() and popover
validity with fully active documents. I also ported the relevent wpt
tests for this change.

Spec change: &lt;<a href="https://github.com/whatwg/html/pull/10705">https://github.com/whatwg/html/pull/10705</a>&gt;

* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-active-document-expected.txt: Rebaseline.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-active-document.html: Rebaseline.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-showModal-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-showModal.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-active-document-expected.txt: Rebaseline.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-active-document.html: Rebaseline.
* Source/WebCore/html/HTMLDialogElement.cpp:
(WebCore::HTMLDialogElement::showModal):
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::checkPopoverValidity):

Canonical link: <a href="https://commits.webkit.org/286020@main">https://commits.webkit.org/286020@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0c400920c6b987f51b5c15d7dabdb462db9f23e1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74301 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53730 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/27112 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78680 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25539 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/76418 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62863 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1515 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58408 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16739 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77368 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48571 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63916 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38820 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45547 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21411 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23872 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66961 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21758 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80197 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1618 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/929 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66701 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1763 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63934 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65981 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16428 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9922 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8075 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1582 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/4370 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1611 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1599 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1618 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

maraisr commented on LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-active-document-expected.txt:
This was the wpt changes for all test changes in this diff.

https://github.com/web-platform-tests/wpt/pull/48854

maraisr commented on LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-active-document.html:
```diff
@@ -0,0 +1,16 @@
+    popover.setAttribute('popover','');
+    doc.body.appendChild(popover);
+    assert_throws_dom('InvalidStateError',() => popover.showPopover());
+    assert_false(popover.matches('[open]'));
```
[The WPT changes, had this as `:open`](https://github.com/web-platform-tests/wpt/pull/48854/files#diff-bc9916a12e7e33a05bdf66e99cccaf6a52a99314c381165ed50c5b8053e26e49R15) — which is not a valid CSSSelector, so the tests failed. So changed it to this:

```diff
-    assert_false(popover.matches(':open'));
+    assert_false(popover.matches('[open]'));
```

Is it worth upstreaming this change as well? If so, I'll action that.

nt1m replied:
I think you want `:popover-open`, `:open` isn't defined on popovers. Popovers also don't set the `[open]` attribute.

maraisr replied:
Thank you for the tip @nt1m — I have pushed that change. 🙌

maraisr committed:
Guard dialog.showModal() and "check popover validity" with fully active checks
https://bugs.webkit.org/show_bug.cgi?id=281550

Reviewed by Darin Adler.

There was a WHATWG/html spec change to now guard showModal() and popover
validity with fully active documents. I also ported the relevent wpt
tests for this change.

Spec change: <https://github.com/whatwg/html/pull/10705>

* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-active-document-expected.txt: Rebaseline.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-active-document.html: Rebaseline.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-showModal-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-showModal.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-active-document-expected.txt: Rebaseline.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-active-document.html: Rebaseline.
* Source/WebCore/html/HTMLDialogElement.cpp:
(WebCore::HTMLDialogElement::showModal):
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::checkPopoverValidity):

Canonical link: https://commits.webkit.org/286020@main
